generator/generator.go

```diff
@@ -0,0 +1,149 @@
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	_ "github.com/docker/machine/drivers/amazonec2"
+	//	_ "github.com/docker/machine/drivers/azure"
+	_ "github.com/docker/machine/drivers/digitalocean"
+	_ "github.com/docker/machine/drivers/exoscale"
+	//	_ "github.com/docker/machine/drivers/google"
+	_ "github.com/docker/machine/drivers/hyperv"
+	_ "github.com/docker/machine/drivers/packet"
+	_ "github.com/docker/machine/drivers/rackspace"
+	_ "github.com/docker/machine/drivers/softlayer"
+	_ "github.com/docker/machine/drivers/virtualbox"
+	_ "github.com/docker/machine/drivers/vmwarefusion"
+	_ "github.com/docker/machine/drivers/vmwarevcloudair"
+	_ "github.com/docker/machine/drivers/vmwarevsphere"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/drivers"
+	"github.com/rancherio/go-machine-service/generator/helpers"
+)
+
+func main() {
+	err := generate()
+	if nil != err {
+		log.Fatal(err)
+	}
+}
+
+func generate() error {
+	resourceData, err := fetchData() // collects all data required to generate the files
+	if nil != err {
+		log.Errorf("error collecting driver data err=%v", err)
+		return err
```

Comment:
Logging and propagating an error is not a great convention. Could you construct a new error (using the message that you're currently logging) and just return that instead?